jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java

```diff
@@ -68,14 +72,101 @@ public FormServiceEntryPointImpl(Instance<FormProvider<? extends FormRenderingSe
         for (FormProvider provider : providersInjected) {
             providers.add(provider);
         }
-
         this.defaultFormProvider = defaultFormProvider;
     }
 
     @Override
     public FormRenderingSettings getFormDisplayTask(String serverTemplateId,
                                                     String domainId,
                                                     long taskId) {
+
+        if (kieServerFormRenderer) {
+            return getKieServerFormDisplayTask(serverTemplateId, domainId, taskId);
+        } else {
+            return getEmbeddedFormDisplayTask(serverTemplateId, domainId, taskId);
+        }
+    }
+
+
+    @Override
+    public FormRenderingSettings getFormDisplayProcess(String serverTemplateId,
+                                                       String domainId,
+                                                       String processId,
+                                                       boolean isDynamic) {
+  
+        if (kieServerFormRenderer) {
+            return getKieServerFormDisplayProcess(serverTemplateId, domainId, processId, isDynamic);
+        } else {
+            return getEmbeddedFormDisplayProcess(serverTemplateId, domainId, processId);
+        }
+    }
+    
+    /*
+     * KieServer based form rendering
+     */
+    
+    protected FormRenderingSettings getKieServerFormDisplayProcess(String serverTemplateId,
```

Comment:
small formatting issue here... overall if you can also remove some of the extra empty lines.